lib/puppet/provider/service/launchd.rb

```diff
@@ -138,7 +138,7 @@ def self.make_label_to_path_map(refresh=false)
         Puppet.debug("Reading launchd plist #{filepath}")
         job = read_plist(filepath)
         next if job.nil?
-        if job.has_key?("Label")
+        if job.is_a?(Hash) and job.has_key?("Label")
```

Comment:
This might be more future-proof as it does not make assumptions on `job` object type
```
job.respond_to?(:key) && job.key?("Label")
```
Also `&&` is preferred over `and` in conditions

Reply:
Done.

Reply:
@ccaviness the last part also needs to be changed from `job.has_key?("Label")` to `job.key?("Label")` to make sure that we use the same method for which we check the existence.

Reply:
Done.